A11y: Replace non-semantic div buttons with <button> in URLPopUpSection

omkarhole opened the pull request:
Replaced non-semantic <div role="button"> elements with proper <button> elements to improve keyboard accessibility, screen reader support, and WCAG compliance. Preserved styling, IDs, aria-labels, and existing behavior.


Changes

- Replaced L.DomUtil.createWithId('div', …) with 'button'
- Removed ARIA roles no longer needed

Preserved:

- IDs
- classes
- aria-labels
- onclick handlers
- icon images
- No visual or behavioral regression

Issue Link:https://github.com/CollaboraOnline/online/issues/12017

@rparth07 @Darshan-upadhyay1110

Darshan-upadhyay1110 commented:
Thanks @omkarhole! I noticed the Change-Id is missing. Please make sure your commits include both a **Signed-off-by (DCO)** line and a **Change-Id**.

### Add Change-Id

To automatically generate a Change-Id:

1. Copy the commit hook:

   ```
   cp .git-hooks/commit-msg .git/hooks/commit-msg
   ```
2. Amend the commit so the Change-Id is added:

   ```
   git commit --amend
   git push -f
   ```

rparth07 reviewed (commented):
Thanks @omkarhole for patch:) 
Overall changes are good! This enhancement will make it better :)

Darshan-upadhyay1110 reviewed (changes requested):
@omkarhole the URL popup UI fully broken now it's not looking okay. A11y should not change the UI look the buttons should be consistent as it was before....

omkarhole commented on browser/css/cool.css:
```diff
@@ -1193,11 +1193,22 @@ body {
 }
 
 .hyperlink-popup-btn {
```
Hii , @rparth07  . I Have Check the changes on my local machine and it dosent shows button UI affected   
<img width="1425" height="218" alt="Screenshot 2026-01-02 113121" src="https://github.com/user-attachments/assets/57e2c407-e013-4a2e-aaff-e8239d659a28" />
<img width="1367" height="175" alt="Screenshot 2026-01-02 112938" src="https://github.com/user-attachments/assets/5531b9a1-13c7-4a9c-bdbd-38c85d4befe7" />  

would you please recheck it again ?

rparth07 replied:
Chromium Version 143.0.7499.146

omkarhole commented:
@Darshan-upadhyay1110  , @rparth07   i have done clean code and make some changes and also created video and now the UI looks Good  . please go through video and check on your local machine also

_video:_

https://github.com/user-attachments/assets/cbfc22a8-aa95-4115-b16b-fda80c3bbc25

Darshan-upadhyay1110 commented:
Please add change ID and DCO. And i see the cypress is not happy can you please check if it's passes the cases in local or not ?

omkarhole committed:
A11y: Replace non-semantic div buttons with <button> in URLPopUpSection

- Replaced div role='button' elements with proper button elements
- Removed redundant ARIA roles (button role is implicit in button elements)
- Fixed aria-label on edit button (was using copyLinkText instead of editLinkText)
- Set alt='' on images since buttons already have aria-labels (avoiding redundancy)
- Added CSS reset for button elements to maintain consistent styling
- Improved keyboard accessibility and screen reader support

Issue: https://github.com/CollaboraOnline/online/issues/12017
Signed-off-by: omkar hole <omkarhole314@gmail.com>
Change-Id: I06f6da85ffbaa656962fad3207c6a4c132d07572

Darshan-upadhyay1110 commented:
Hi @omkarhole do you think you can rebase and work on what parth ask to improve in this PR ? i am clearing long hanging PR's so do respond here so we can close/merge this patch soon...

Darshan-upadhyay1110 reviewed (commented):
Hi @omkarhole do you think you can rebase and work on what parth ask to improve in this PR ? i am clearing long hanging PR's so do respond here so we can close/merge this patch soon...

rparth07 commented on browser/css/cool.css:
```diff
@@ -1198,11 +1193,25 @@ body {
-	display: inline-block;
-	margin-left: 12px;
+	display: inline-flex;
+	align-items: center;
+	justify-content: center;
 	vertical-align: middle;
+	margin-left: 12px;
 	width: 26px;
 	height: 26px;
+	/* Reset default native button styles */
+	-webkit-appearance: none;
+	-moz-appearance: none;
+	appearance: none;
+	background: none;
+	border: none;
+	padding: 0;
+	box-shadow: none;
+	cursor: pointer;
+	font: inherit;
+	color: inherit;
+	outline: none;
```
Instead of this, you might want to add the relevant class in `browser/css/btns.css`. i.e. see recent PR where I added the class to remove the button layout https://github.com/CollaboraOnline/online/pull/15347/changes#diff-cd3b6e2a746b0e123774fc6002246a28dd790fbe82a0f88e48348075e7945845.

also check `browser/css/device-desktop.css` to remove it from hover effect. Then I believe we just need existing css + background and border css properties only.

rparth07 commented:
Button UI is still visible.
<img width="388" height="94" alt="image" src="https://github.com/user-attachments/assets/d2abd49b-41cc-4eaa-82d5-343bc7de4f63" />

Darshan-upadhyay1110 commented:
Hi, could you please move this PR to Gerrit? [We have now transitioned to Gerrit for code reviews](https://forum.collaboraonline.com/t/your-first-pull-request/41), and I'm currently cleaning up the remaining GitHub PRs.

I'll go ahead and close this PR for now, but we'd be happy to continue the review once it's submitted through Gerrit. Thanks!
